packages/node_modules/@webex/redux-module-media/src/actions.js

```diff
@@ -217,67 +244,66 @@ export function listenForCalls(sparkInstance) {
+    meetingPromise = new Promise((resolve) => {
+      meetingsAdapter.createMeeting(destination)
+        .pipe(first())
+        .subscribe(({ID}) =>
+          resolve(meetingsAdapter.datasource.meetings.getMeetingByType('id', ID)));
+    });
 
-    // Reference for device.kind: https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/enumerateDevices
-    return navigator.mediaDevices.enumerateDevices()
-      .then((devices) => ({
-        audio: devices.filter((device) => device.kind === 'audioinput').length > 0,
-        video: devices.filter((device) => device.kind === 'videoinput').length > 0
-      }));
-  });
-}
+    return meetingPromise;
+  }
+
+  return meetingPromise
+    .then(tap((meeting) => meetingsAdapter.datasource.meetings.meetingCollection.set(meeting)))
+    .then((meeting) => meetingsAdapter.getLocalMedia(meeting.id)
+      .then(({localAudio, localVideo}) => meetingsAdapter.fetchMeetingTitle(destination)
+        .then((title) => {
+          // eslint-disable-next-line no-param-reassign
+          meetingsAdapter.meetings[meeting.id] = {
+            ID: meeting.id,
+            title,
+            localAudio,
+            localVideo,
+            localShare: null,
+            remoteAudio: null,
+            remoteVideo: null,
+            remoteShare: null
+          };
+
+          return meeting;
+        })));
+};
 
 /**
  * Call a user with an email address or userId
  *
  * @export
- * @param {Object} sparkInstance
+ * @param {Object} sdkAdapter
  * @param {String} data.destination
- * @param {Object} data.options
  * @returns {Promise}
  */
-export function placeCall(sparkInstance, {destination, options}) {
-  return (dispatch) => {
-    let call, id;
-
-    return getSupportedDevice()
-      .then(({audio, video}) => {
-        try {
-          call = sparkInstance.phone.dial(
-            destination,
-            Object.assign(
-              options,
-              {
-                offerOptions: {offerToReceiveVideo: true, offerToReceiveAudio: true},
-                audio,
-                video
-              }
-            )
-          );
-          ({call, id} = processCall({dispatch, call}));
-        }
-        catch (e) {
-          // Ignore, will be caught on call.on('error')
-        }
-
-        return {call, id};
-      });
-  };
+export function placeCall(sdkAdapter, {destination, cleanUp}) {
+  const {meetingsAdapter} = sdkAdapter;
+
+  return (dispatch) => getMeeting({sdkAdapter, destination})
+    .then(tap((meeting) => processCall({dispatch, call: {...meeting}, destination})))
+    .then((meeting) => {
+      const bye = meeting.leave;
+
+      // eslint-disable-next-line no-param-reassign
+      meeting.leave = (...args) => bye.bind(meeting)(...args)
+        .then(() => dispatch(removeCall(meeting.id)))
+        .then(() => dispatch(cleanUp()))
+        // eslint-disable-next-line no-param-reassign
+        .finally(() => { meeting.leave = bye.bind(meeting); });
```

Comment:
Does this work?

```suggestion
        .finally(() => (meeting.leave = bye.bind(meeting)));
```

Reply:
Yes it is the same as:
```suggestion
        .finally(() => {
          meeting.leave = bye.bind(meeting);
        });
```
It's a function that does one assignment.